### ISSUE TITLE
Dev Work Only: Engage filter logic by account type - allowance sub types 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#2230]
### Definition of Ready
- [x] Does this meet the INVEST criteria?
- [x] Does this have a clearly defined user?
- [x] Does this have an estimate?
- [x] Does this have acceptance criteria?
- [x] Does this have sufficient context for the specific functionality being developed (design, business rules, etc.)
- [x] Does this have a dev link been provided in the design?
- [x] Is this traceable to a feature, epic, or PI?

## User Story
As a custom data download user I want the filter logic to function for account type So that I can avoid producing an allowance query with no results

## Acceptance Criteria
Given that a user has applied account info or allowance holdings as a data sub type
When the filter panel functionality is displayed
Then the filter logic is engaged

Given that the filter logic is engaged for allowance transactions 
When filter criteria is applied for account type 
Then all other filters are narrowed based on the applied account type criteria

Given that a user has selected an allowance sub type (allowance holdings and account info)
When filter criteria is applied for account type 
Then all other filters are narrowed based on the applied account type criteria

Given that the filter logic is engaged (allowance holdings, account info, allowance transactions)
When filter criteria is applied 
Then the account type filter is narrowed based on the applied criteria

Given that the filter logic is engaged
When new filter criteria is applied
Then all previously selected criteria that are still valid are maintained

Given that the filter logic is engaged
When the filter panel functionality is displayed
Then non applicable selections are removed from the combo box filter list
AND grayed out from the check lists

Given that the filter logic is engaged
When account type criteria is removed (ex filter tags, filters)
Then the filters are narrowed based on the applied criteria

Given that the filer logic is engaged
When the the checklist functionality is displayed
Then the user can not select or apply filter criteria for greyed out items

Given that the filter logic is engaged
When the "select all" functionality is selected on the checklist component
Then the system selects criteria that is not greyed out

Given that the filter logic is engaged
When the checklist functionality is displayed
Then the select all functionality is disabled for groupings that only contain greyed out criteria

### Development Notes
- address account type filter logic for account info, allowance holdings, allowance transactions 

- Will need to switch over to using account type code instead of account type 

- The scope of this ticket includes the following: 
1. Narrow other filters based on applied account type criteria 
2. Narrow the account type filter based on applied criteria 

### Does this Story require 508 Testing?
- [ ] Implementing a new authoring tool that will produce code / content for the application
- [ ] Implementing a 3rd party widget that will be part of the application
- [ ] Adding new / updating custom functionality 
- [ ] Adding new / updating a page template
- [ ] Adding new / replacing existing functionality (unless the replacement has already passed 508 testing)
- [ ] Full Site Review prior to a scheduled EPA 508 review
- [ ] Planning a video that will appear on the application
- [ ] Producing a video that will appear on the application
- [ ] Producing Training materials
- [ ] Producing documents

## Context
(User journeys, design mockups, architectural diagrams, business rules, data element definitions)

Dependency:  
1. Will wait on the ERG to make db changes to work on this ticket.
2. Will need the script that the ERG ran to make the db changes 

### Definition of Done
- [ ] Have all Acceptance Criteria been agreed to and validated by the scrum team?
- [ ] Has the code been unit tested and peer reviewed?
- [ ] Have the functional tests been executed?
- [ ] Have all defects been reviewed and dispositioned, resolved, or deferred?
- [ ] Is the user story ready to be deployed to the test/staging environments?

**Comment by JanellC** [comment]
@tlageman-epa Development work is complete. Adjusted the ticket estimate from 3 to a 2. Manual testing will take place with ticket Test Only: Engage Filter Logic By Account type- allowance sub types #2506
